doc/neps/nep-0052-python-api-cleanup.rst

```diff
@@ -160,16 +202,65 @@ Details TBD, something like::
     numpy.testing
     numpy.typing
 
-    # Special-purpose
+    # Special-purpose namespaces. Keep these, but document them in a separate
+    # grouping in the reference guide and explain their purpose.
     numpy.array_api
     numpy.ctypeslib
     numpy.emath
-    numpy.f2py
+    numpy.f2py  # only a couple of public functions, like `compile` and `get_include`
     numpy.math
     numpy.lib.stride_tricks
     numpy.rec
     numpy.types
 
+    # Legacy (prefer not to use). Third grouping in the reference guide.
+    numpy.char
+    numpy.distutils
+    numpy.matrixlib  # or deprecate?
+
+    # To remove
+    numpy.compat
+    numpy.core?
```

Comment:
Where would core functions go? Perhaps rename to `_core`.

Reply:
Yes, that seems like the obvious choice to me.